src/web/ui/Chrome.js

```diff
@@ -476,19 +473,12 @@ exports = Class(CenterLayout, function (supr) {
 
   this._setViewport = function (size, viewport, scale) {
     $.style(this.contents, sizeToCSS(size, scale));
-    // $.style(this.frameWrapper, sizeToCSS(viewport, scale));
     if (this._frame) {
-      var style;
-
-      if (this._isRetina) {
-        style = sizeToCSS(viewport);
-        style[TRANSFORM_STYLE] = 'scale(' + scale + ')';
-        style[TRANSFORM_ORIGIN_STYLE] = '0px 0px';
-      } else {
-        style = sizeToCSS(viewport, scale);
-        style[TRANSFORM_STYLE] = '';
-      }
-
+      var dpr = this.getDevicePixelRatio();
+      var style = sizeToCSS(viewport, 1 / dpr);
+      var scaleDPR = dpr * scale;
```

Comment:
Specifically, I think we could replace `this._setViewport(size, viewportSize, scale)` can be `this._setViewport(size, viewportSize, zoom)`?